src/searchresultslistformaction.cpp

```diff
@@ -43,10 +61,43 @@ bool SearchResultsListFormAction::process_operation(
 			v->get_statusline().show_message(_("Already in first search result."));
 		}
 		break;
+	case OP_RELOAD:
+		v->get_statusline().show_error(
+			_("Error: you can't reload search results."));
+		break;
 	default:
 		return ItemListFormAction::process_operation(op, automatic, args);
 	}
 	return true;
 }
 
+void SearchResultsListFormAction::set_head(const std::string& s,
+	unsigned int unread,
+	unsigned int total,
+	const std::string& url)
+{
+	std::string title;
+	FmtStrFormatter fmt;
+
+	fmt.register_fmt('N', PROGRAM_NAME);
+	fmt.register_fmt('V', utils::program_version());
+
+	fmt.register_fmt('u', std::to_string(unread));
+	fmt.register_fmt('t', std::to_string(total));
+
+	auto feedtitle = s;
+	utils::remove_soft_hyphens(feedtitle);
+	fmt.register_fmt('T', feedtitle);
+
+	fmt.register_fmt('U', utils::censor_url(url));
+
+	fmt.register_fmt('F', apply_filter ? matcher.get_expression() : "");
+
+	const unsigned int width = utils::to_u(f.get("title:w"));
+	title = fmt.do_format(
+			cfg->get_configvalue("searchresult-title-format"),
+			width);
+	set_value("head", title);
+}
```

Comment:
Instead of repeating all this code here, let's put it into a helper method in the base class. The method would take all the same arguments as this one, plus the title format string. Then `set_head()` methods turn into a single call of that helper method.

Reply:
@Minoru Of course, I'm not that well versed in the project. But I think that the current situation indicates problems in architecture and design. And the solution proposed by you is not a way out, but only a temporary patch. I'd rather think more. That's what I'm busy with right now.

Reply:
@Minoru So, I tried to split the `set_head` method into 2 parts, the first part setup `FmtStrFormatter`, the second part does everything else.
I moved the setup of `FmtStrFormatter` into a separate method `ItemListFormAction::setup_head_formatter`, I call it inside `ItemListFormAction::set_head` and `SearchResultsListFormAction::set_head`, which is overridden.
`FmtStrFormatter` pass inside `setup_head_formatter` by reference, because for some reason `FmtStrFormatter` delete copy constructor.(why?)

This solution reduces code duplication.

But we also have `set_head` method inside `ItemViewFormAction` and it works in a similar way. We could use `setup_head_formatter` here too.

How about this solution?

Reply:
@Minoru ok, I made a commit with this changes. If something wrong, `git revert` exists)))

Reply:
I don't see how this solution is any different from the one I proposed, but it's fine, let's do it this way then.

I wouldn't change `ItemViewFormAction::set_head()`, because there's nothing to be gained from splitting that method: there are no derived classes to reuse the code, and the method isn't so large that it warrants splitting.

> for some reason FmtStrFormatter delete copy constructor.(why?)

The short answer is: because no-one bothered to write a copy constructor. The long answer is: `FmtStrFormatter` is a wrapper over pointer to a Rust struct. It acts much like an `std::unique_ptr`, which is why it doesn't have a copy constructor. It's possible to implement one, but you'll have to write some Rust code to call `clone()` on the Rust struct and put it behind a new pointer.

It looks like `FmtStrFormatter` can be made movable though. Please try that.

Reply:
> It looks like FmtStrFormatter can be made movable though. Please try that.
I tried, it didn't work, and then I thought it's overcomplicated. Pass by reference is simple and just work, like KISS principle.

Reply:
An out-parameter is more complicated (and less natural) than a just returning a value. I just pushed the commit that implements it.